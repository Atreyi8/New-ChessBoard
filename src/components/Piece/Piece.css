.piece {
    font-size: 24px;
    font-weight: bold;
  }
  
  .piece.black {
    color: black;
  }
  
  .piece.white {
    color: white;
  }
  