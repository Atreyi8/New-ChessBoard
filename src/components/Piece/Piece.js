import React from "react";
import "./Piece.css";

const Piece = ({ symbol, color }) => {
  return <span className={`piece ${color}`}>{symbol}</span>;
};

export default Piece;
