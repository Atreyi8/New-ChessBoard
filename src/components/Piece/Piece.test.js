import React from "react";
import { render } from "@testing-library/react";
import Piece from "./Piece";

describe("Piece Component", () => {
  it("renders correctly with given props", () => {
    const { container } = render(<Piece symbol="â™”" color="white" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with an empty symbol", () => {
    const { container } = render(<Piece symbol="" color="black" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
