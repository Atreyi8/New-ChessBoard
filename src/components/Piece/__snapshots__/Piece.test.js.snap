// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Piece Component renders correctly with an empty symbol 1`] = `
<span
  class="piece black"
/>
`;

exports[`Piece Component renders correctly with given props 1`] = `
<span
  class="piece white"
>
  â™”
</span>
`;
